slack.go

```diff
@@ -85,12 +108,50 @@ func listenToSlackEvents(client *socketmode.Client, api *slack.Client, config *c
 
 			switch callback.Type {
 			case slack.InteractionTypeBlockActions:
-				go func() {
-					if err := runCodefreshPipeline(api, config, callback); err != nil {
-						logger.Errorw("unable to run codefresh pipeline",
-							"error", err, "raw", callback)
+
+				if callback.BlockActionState == nil {
+					// we need the state of the action to know what to do
+					// with it, else, we drop the message
+					logger.Errorw("unable to process interactive message",
+						"error", "no block_action state field", "raw", callback)
+					continue
+				}
+
+				// from here, it is safe to call BlockActionState
+				actions := callback.BlockActionState.Values
+				for id, action := range actions {
+					switch id {
+
+					case SlackTriggerTechAllyProject:
+						repo := action[SlackSelectedTechAllyProject].SelectedOption.Value
+						go func() {
+							if err := runCodefreshPipeline(api, config, callback, repo); err != nil {
+								logger.Errorw("unable to run codefresh pipeline",
```

Comment:
nit: upcase Codefresh
```suggestion
								logger.Errorw("unable to run Codefresh pipeline",
```